OpenSim/Common/DelimFileAdapter.h

```diff
@@ -400,41 +403,51 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
                      _timeColumnLabel,
                      column_labels[0]);
     column_labels.erase(column_labels.begin());
-    // Set the column labels as metadata.
-    ValueArray<std::string> value_array{};
-    for(const auto& cl : column_labels)
-        value_array.upd().push_back(SimTK::Value<std::string>{cl});
-    typename TimeSeriesTable_<T>::DependentsMetaData dep_metadata{};
-    dep_metadata.setValueArrayForKey("labels", value_array);
-    table->setDependentsMetaData(dep_metadata);
 
     // Read the rows one at a time and fill up the time column container and
     // the data container.
+    std::vector<double> timeVec;
+    std::vector<SimTK::RowVector_<T>> depDataVec;
+    int initCapacity = 2500;
+    timeVec.reserve(initCapacity);
+    depDataVec.reserve(initCapacity);
     auto row = nextLine();
-    while(!row.empty()) {
-        ++line_num;
 
+    while (!row.empty()) {
+        ++line_num;
+        
         // Time is column 0.
-        double time = std::stod(row.front());
+        timeVec.push_back(std::stod(row.front()));
         row.erase(row.begin());
 
         auto row_vector = readElems(row);
 
         OPENSIM_THROW_IF(row_vector.size() != column_labels.size(),
-                         RowLengthMismatch,
-                         fileName,
-                         line_num,
-                         column_labels.size(),
-                         static_cast<size_t>(row_vector.size()));
-
-        // Column 1 is time.
-        table->appendRow(time, std::move(row_vector));
+            RowLengthMismatch,
+            fileName,
+            line_num,
+            column_labels.size(),
+            static_cast<size_t>(row_vector.size()));
+        
+        depDataVec.push_back(row_vector);
 
         row = nextLine();
     }
 
+    // Create and fill up the matrix of data
+    int nrow = static_cast<int>(depDataVec.size());
+    int ncol = static_cast<int>(column_labels.size());
+    SimTK::Matrix_<T> matrix(nrow, ncol);
+    for (int iRow = 0; iRow < static_cast<int>(depDataVec.size()); ++iRow) {
+        matrix.updRow(iRow) = depDataVec[iRow];
+    }
+
+    // Create the table and update other metadata from above
+    auto& table = *new TimeSeriesTable_<T>(timeVec, matrix, column_labels);
```

Comment:
Please ignore this comment: Ideally, we would `std::move` the matrix into the table, but this would require changing the signature of the table constructor to take a `T&&`   or `T` intead of a `const T&`.